doc/source/how-to-upgrade-to-flower-next.rst

```diff
@@ -254,62 +316,120 @@ Simulation in CLI
 
-    # Flower 1.7
+    # Flower v1.7
     $ python sim.py
 
-- Set default resources for each |clientapp_link|_ using the ``--backend-config``
-  command line argument instead of setting the ``client_resources`` argument in
-  |startsim_link|_. Here's an example:
+Depending on your Flower version, you can also define the default resources as follows:
+
+- for Flower versions 1.11 and onwards, you can edit your pyproject.toml file and then
+  run ``flwr run`` in CLI as shown in the example below.
+- for Flower versions between 1.8 to 1.10, you can adjust the resources for each
+  |clientapp_link|_ using the ``--backend-config`` command line argument instead of
+  setting the ``client_resources`` argument in |startsim_link|_.
+- for Flower versions before 1.8, you need to run |startsim_link|_ and pass a dictionary
+  of the required resources to the ``client_resources`` argument.
 
 .. code-block:: bash
-    :emphasize-lines: 6
+    :emphasize-lines: 2,8
+
+    # Flower v.1.11+
+    # pyproject.toml
+    [tool.flwr.federations.local-sim-gpu]
+    options.num-supernodes = 10
+    options.backend.client-resources.num-cpus = 2
+    options.backend.client-resources.num-gpus = 0.25
 
-    # Flower 1.8
+    $ flwr run
+
+    # Flower v1.8 - v1.10
     $ flower-simulation \
         --client-app=sim:client_app \
         --server-app=sim:server_app \
-        --num-supernodes=100 \
+        --num-supernodes=10 \
         --backend-config='{"client_resources": {"num_cpus": 2, "num_gpus": 0.25}}'
 
-.. code-block:: python
-    :emphasize-lines: 5
-
-    # Flower 1.7 (in `sim.py`)
+    # Flower v1.7 (in `sim.py`)
     if __name__ == "__main__":
         hist = flwr.simulation.start_simulation(
-            num_clients=100, client_resources={"num_cpus": 2, "num_gpus": 0.25}, ...
+            num_clients=10, client_resources={"num_cpus": 2, "num_gpus": 0.25}, ...
         )
 
 Simulation in a Notebook
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-- Run |runsim_link|_ in your notebook instead of |startsim_link|_. Here's an example:
+To run your simulation from within a notebook, please consider the following examples
+depending on your Flower version:
+
+- for Flower versions 1.11 and onwards, you first need to edit your pyproject.toml file
+  as shown in the example below and run |runsim_link|_ in your notebook instead of
+  |startsim_link|_.
+- for Flower versions between 1.8 to 1.10, you need to run |runsim_link|_ in your
+  notebook instead of |startsim_link|_ and configure the resources.
+- for Flower versions before 1.8, you need to run |startsim_link|_ and pass a dictionary
+  of the required resources to the ``client_resources`` argument.
 
 .. code-block:: python
-    :emphasize-lines: 19,27
+    :emphasize-lines: 2,9,13,17,21
+
+    # Flower v1.11+
+    # pyproject.toml
+    [tool.flwr.federations.local - sim - gpu]
+    options.num - supernodes = 10
+    options.backend.client - resources.num - cpus = 2
+    options.backend.client - resources.num - gpus = 0.25
```

Comment:
```suggestion
```

Reply:
This should be removed as there is no `pyproject.toml` when doing things in a notebook environment

Reply:
Thank you for pointing this out. Removed. I thought users need to define the configurations in the `pyproject.toml` before starting the notebook, but they can do so from within the notebook.